torch/compiler/__init__.py

```diff
@@ -660,4 +660,4 @@ def load_compiled_function(file: io.IOBase):
     from torch._dynamo.aot_compile import CompileArtifacts
 
     data = file.read()
-    return CompileArtifacts.deserialize(data).compiled_function()
+    return CompileArtifacts.deserialize(data)
```

Comment:
Right now I consider CompileArtifacts an impl detail and not sure we will run into issue if user can access its data directly later. I'm fine making it callable but it seems now we expose strictly more information from load_compiled_function (and .aot_compile())

could we return a wrapper that is callable-only every time we pass compiled function to user space (also to match what dynamo does today)?

Another option is to type annotation the return type as `Callable` so user might have an idea it's callable-only (but I don't prefer this since no one reads type annotation in python :( )